deploy/transport-controller-deployment.yaml.template

```diff
@@ -0,0 +1,89 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: transport-controller-config
+  namespace: ${WDS_NAME}-system
+data:
+  get-kubeconfig.sh: |
+    #!/bin/bash
+    # this script receives a ControlPlane name and a parameter
+    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
+    # or the external kubeconfig (if set to "true", the first will be retrieved).
+    # The function returns the requested kubeconfig's content in base64.
+    # it assumes that it runs within a kubeflex context.
+
+    cpname="$1"
+    get_incluster_key="$2"
+
+    key=""
+    if [[ $get_incluster_key == "true" ]]; then
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
+    else
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+    fi
+
+    # get secret details
+    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
+
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: transport-controller
+  namespace: ${WDS_NAME}-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: transport-controller
+  template:
+    metadata:
+      labels:
+        name: transport-controller
+    spec:
+      serviceAccountName: kubestellar-controller-manager
+      initContainers:
+      - name: setup-wds-kubeconfig
+        image: bitnami/kubectl
```

Comment:
In the other repos we use our custom quay.io/kubestellar/kubect images because of the bandwidth limitations in using DockerHub images. Can we agree on what to use for all the repos?

Reply:
good point, sure. 
I was not aware that we have kubectl image under kubestellar.
of course just need to validate it works as expected.
pushed a fix (for both comments).

Reply:
you may have to specify a tab/version since we don't have latest/stable
`quay.io/kubestellar/kubectl:1.27.8`

Reply:
I think there was also some issues with some of the bitnami images not being multi arch, so in any case it is better to use our own

Reply:
looks like quay.io/kubestellar/kubectl image is working different than bitnami. 
probably a minor difference, but it broke the init container setup.
will debug and push a fix soon

Reply:
seems like `quay.io/kubestellar/kubectl` doesn't have bash.

Reply:
pushed a fix. will see if e2e works now

Reply:
e2e are working with quay.io/kubestellar/kubectl image. resolving thread.